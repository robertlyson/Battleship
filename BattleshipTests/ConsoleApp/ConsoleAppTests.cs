using ApprovalTests;
using ApprovalTests.Reporters;
using BattleshipGame.Games;
using NUnit.Framework;
using Shouldly;
using Spectre.Console.Testing;

namespace BattleshipTests.ConsoleApp;

public class ConsoleAppTests
{
    [Test]
    public void Quit_game()
    {
        var console = new TestConsole();
        console.Input.PushTextWithEnter("A2");
        console.Input.PushTextWithEnter("q");

        new App(console).Start(GameFixture());

        var actual = console.Output;
        Approvals.Verify(actual);
    }

    [Test]
    public void Play_game_with_single_ship()
    {
        var console = new TestConsole();
        console.Input.PushTextWithEnter("A1");

        var game = new Game(new Ship[]
        {
            new SinglePositionShip("Foo", new Position('A', 1))
        });
        new App(console).Start(game);

        var actual = console.Output;
        Approvals.Verify(actual);
    }

    private static Game GameFixture()
    {
        return new Game(new Ship[] { new SinglePositionShip("Foo", new Position('A', 1)) });
    }
}