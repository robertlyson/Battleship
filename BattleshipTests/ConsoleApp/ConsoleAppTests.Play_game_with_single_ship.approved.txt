                  BATTLESHIP                  
┌────┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐
│    │ A │ B │ C │ D │ E │ F │ G │ H │ I │ J │
├────┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┤
│ 1  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 2  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 3  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 4  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 5  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 6  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 7  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 8  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 9  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 10 │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
└────┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┘
General, which position (e.g. A1) to attack(q to exit game)? A1
[2J[3J[1;1H                  BATTLESHIP                  
┌────┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐
│    │ A │ B │ C │ D │ E │ F │ G │ H │ I │ J │
├────┼───┼───┼───┼───┼───┼───┼───┼───┼───┼───┤
│ 1  │ x │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 2  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 3  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 4  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 5  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 6  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 7  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 8  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 9  │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
│ 10 │ o │ o │ o │ o │ o │ o │ o │ o │ o │ o │
└────┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┘
Attack on A1 sunk Foo ship.
Game Over.